pkg/tfvars/openstack/openstack.go

```diff
@@ -101,24 +105,15 @@ func TFVars(
 		userManagedLoadBalancer = true
 	}
 
-	var zones []string
-	{
-		seen := make(map[string]struct{})
-		for _, config := range masterSpecs {
-			if _, ok := seen[config.AvailabilityZone]; !ok {
-				zones = append(zones, config.AvailabilityZone)
-				seen[config.AvailabilityZone] = struct{}{}
-			}
-		}
+	computeAvailabilityZones := make([]string, len(masterSpecs))
+	for i := range computeAvailabilityZones {
+		computeAvailabilityZones[i] = masterSpecs[i].AvailabilityZone
```

Comment:
We used to pick the availability zone from the install-config. Instead, this function now takes zones from machine generation. No need to duplicate logic here.

The interesting side effect is that `zones` is no longer the list of the available zones: it's now a list where each index corresponds to a specific machine. This means less computation on the Terraform side.